src/data/actions/assets.js

```diff
@@ -121,46 +121,44 @@ export const uploadingAssets = count => ({
   count,
 });
 
-export const uploadAssetSuccess = (asset, response) => ({
+export const uploadAssetSuccess = response => ({
   type: assetActions.UPLOAD_ASSET_SUCCESS,
-  asset,
   response,
 });
 
-export const uploadAssetFailure = (file, response) => ({
+export const uploadAssetFailure = (asset, response) => ({
   type: assetActions.UPLOAD_ASSET_FAILURE,
-  file,
+  asset,
   response,
 });
 
 export const uploadAssets = (files, courseDetails) =>
   (dispatch, getState) => {
     dispatch(uploadingAssets(files.length));
-    files.forEach((file) => {
+    // gather all the promises into a single promise that can be returned
+    return Promise.all(files.map(file => (
       clientApi.postUploadAsset(courseDetails.id, file)
         .then((response) => {
           if (response.ok) {
-            response.json().then((data) => {
-              dispatch(uploadAssetSuccess(data.asset, response));
-              dispatch(getAssets(getState().request, courseDetails));
+            return response.json().then((json) => {
+              dispatch(uploadAssetSuccess(json));
+              // dispatch(getAssets(..)) returns a promise, so we return it
+              // up to chain it
+              return dispatch(getAssets(getState().request, courseDetails));
             });
-          } else {
-            dispatch(uploadAssetFailure(file, response));
           }
-        });
-    });
+          dispatch(uploadAssetFailure(file, response.status));
```

Comment:
This has been moved out of the else case, since the else case was [unnecessary](https://eslint.org/docs/rules/no-else-return):

> Unnecessary 'else' after 'return'. (no-else-return)